src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutsIndexValidator.cs

```diff
@@ -75,19 +89,33 @@ public void LogWarningIfTimeoutEntityIndexIsIncorrect()
             }
         }
 
-        private IIndex GetIndex(DbConnection connection, PersistentClass entity, string indexName)
+        IIndex GetIndex(DbConnection connection, PersistentClass entity, string indexName)
         {
-
-
             // Check if running on SQL Server.
             if (typeof(MsSql2005Dialect).IsAssignableFrom(dialect.GetType()))
             {
-                var restrictions = new string[5]
+                string EnsureUnquoted(string name)
                 {
-                    entity.Table.Catalog,
-                    entity.Table.Schema,
-                    entity.Table.Name,
-                    indexName,
+                    if (null == name || name.Length < 2)
```

Comment:
I'm concerned there could be a hidden bug in all this math and substringing. Did you consider `Regex.Replace(name, @"[\[\]]+", "")`? (Just removes any `[` or `]` anywhere within the string.) Or does it have to be only on the boundaries to be removed?

Reply:
So what this does is the following:
  * It **does not** validate if the identifier is correct
  * If the identifier is wrapped in `[]` it assumes that it's quoted and:
   * Removes the wrapping `[]` brackets
   * Replaces all double right brackets `]]` with a single one `]`. This is needed because right brackets need to be escaped in quoted identifiers.

This is a less generic version of (unquote) [https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.SqlServer/Addressing/NameHelper.cs#L17] logic in SqlServer transport.

Reply:
:+1: I meant it as a curiosity more than anything, no need to change.